### PR TITLE
fix(pubsub): add commit schema and list schema revision samples

### PR DESCRIPTION
**Pull request by alevenberg**
Issue #11720

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11840)

### CODE REVIEW

**Review by coryan — APPROVED** [review]

File: google/cloud/pubsub/samples/samples.cc [hunk 1]
```diff
@@ -834,6 +835,65 @@ void CreateProtobufSchema(google::cloud::pubsub::SchemaServiceClient client,
   (std::move(client), argv.at(0), argv.at(1), argv.at(2));
 }
 
+void CommitAvroSchema(google::cloud::pubsub::SchemaServiceClient client,
+                      std::vector<std::string> const& argv) {
+  //! [START pubsub_commit_avro_schema] [commit-avro-schema]
```
Comment (dbolduc) [inline google/cloud/pubsub/samples/samples.cc:840]: FYI: The second tag is unnecessary and can be removed.  (unless you plan to write a doc that links it). Here's why...

---

There are two types of tags for samples.

The first tag `[START pubsub_commit_avro_schema]` is used by the DevRel teams. These tags denote samples that show up on https://cloud.google.com/pubsub/docs/samples (for example).

The second tag, `[commit-avro-schema]`, is something our team uses to include samples in our documentation.

For example, the `google::cloud::pubsub::Publisher`:

https://github.com/googleapis/google-cloud-cpp/blob/facd5b44ec871cb4a40985615315c024e88d6835/google/cloud/pubsub/publisher.h#L75

.... links to this sample:

https://github.com/googleapis/google-cloud-cpp/blob/facd5b44ec871cb4a40985615315c024e88d6835/google/cloud/pubsub/samples/samples.cc#L1400-L1439

... which results in the following documentation:

https://cloud.google.com/cpp/docs/reference/pubsub/latest/classgoogle_1_1cloud_1_1pubsub_1_1Publisher#example-using-a-custom-thread-pool

**Review by dbolduc — COMMENTED** [review]
nit: This type of PR should be a `doc(pubsub)` instead of a `fix(pubsub)`.

(Oops, I am too late on the review. It's not a big deal. Just use `doc` for the next one).

File: google/cloud/pubsub/samples/samples.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/samples/samples.cc:840]: Ah i see will send a PR to fix

File: google/cloud/pubsub/samples/samples.cc [hunk 2]
```diff
@@ -1917,6 +1999,9 @@ void AutoRunAvro(
   std::cout << "\nRunning ListSchemas() sample" << std::endl;
   ListSchemas(schema_admin, {project_id});
 
+  std::cout << "\nRunning ListSchemaRevisions() sample" << std::endl;
```
Comment (0000854453) [inline google/cloud/pubsub/samples/samples.cc:2002]: @rebase

**Review by 0000854453 — COMMENTED** [review]
[]()

**Comment by 0000854453** [comment]
> []()

**Pull request merged**